configuration.go

```diff
@@ -0,0 +1,85 @@
+package caddy
+
+import (
+	"github.com/darkweak/souin/configurationtypes"
+	"go.uber.org/zap"
+)
+
+// DefaultCache the struct
+type DefaultCache struct {
+	Distributed bool
+	Headers     []string
+	Olric       configurationtypes.CacheProvider
+	Regex       configurationtypes.Regex
+	TTL         string
+}
+
+// GetDistributed returns if it uses Olric or not as provider
+func (d *DefaultCache) GetDistributed() bool {
+	return d.Distributed
+}
+
+// GetHeaders returns the default headers that should be cached
+func (d *DefaultCache) GetHeaders() []string {
+	return d.Headers
+}
+
+// GetOlric returns olric configuration
+func (d *DefaultCache) GetOlric() configurationtypes.CacheProvider {
+	return d.Olric
+}
+
+// GetRegex returns the regex that shouldn't be cached
+func (d *DefaultCache) GetRegex() configurationtypes.Regex {
+	return d.Regex
+}
+
+// GetTTL returns the default TTL
+func (d *DefaultCache) GetTTL() string {
+	return d.TTL
+}
+
+//Configuration holder
+type Configuration struct {
+	DefaultCache *DefaultCache
+	API          configurationtypes.API
+	URLs         map[string]configurationtypes.URL
+	LogLevel     string
+	logger       *zap.Logger
+	Ykeys        map[string]configurationtypes.YKey
```

Comment:
It's not very important for now, but IMHO we shouldn't blindly copy the Varnish API and expose something modern and consistent with the rest of Caddy's APIs.

Reply:
We have to be compliant to be able to switch from Varnish to Souin without refactoring anything (e.g. for API P).

Reply:
IMHO compatibility with the Varnish API is not a goal of this module (as compatibility with NGINX isn't a goal of Caddy). We could provide an adapter, but the native API should be in sync with current modern best practices (and the name of this header is very weird).

Reply:
IMHO Souin has to be compliant. If it's not mandatory in Caddy we can just remove it from the documentation.

Reply:
What about this? 
Should I just exclude the Ykeys support for caddy? 
Just hide this from this caddy-module documentation and let the users free of enabling it?